src/Lean/Linter/Basic.lean

```diff
@@ -5,18 +5,71 @@ Authors: Lars König
 -/
 prelude
 import Lean.Data.Options
+import Lean.MonadEnv
 import Lean.Log
 
 namespace Lean.Linter
 
+/-- Linter sets are represented as a map from linter name to set name,
+to make it easy to look up which sets to check for enabling a linter.
+-/
+def LinterSets := NameMap (Array Name)
+  deriving EmptyCollection, Inhabited
+
+/-- Insert a set into a `LinterSets` map.
+
+`entry.1` is the name of the linter set,
+`entry.2` contains the names of the set's linter options.
+-/
+def insertLinterSetEntry (map : LinterSets) (entry : Name × Array Name) : LinterSets :=
```

Comment:
I'd prefer just have two separate arguments here, even if you have to (un?)curry at the call site for `addEntry`.